iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/pipe/task/AlterPipeProcedureV2.java

```diff
@@ -0,0 +1,344 @@
+              .getConsensusManager()
+              .write(new AlterPipePlanV2(currentPipeStaticMeta, currentPipeRuntimeMeta));
+    } catch (ConsensusException e) {
+      LOGGER.warn("Failed in the write API executing the consensus layer due to: ", e);
+      response = new TSStatus(TSStatusCode.EXECUTE_STATEMENT_ERROR.getStatusCode());
+      response.setMessage(e.getMessage());
+    }
+    if (response.getCode() != TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+      throw new PipeException(response.getMessage());
+    }
+  }
+
+  @Override
+  protected void rollbackFromOperateOnDataNodes(ConfigNodeProcedureEnv env) throws IOException {
+    LOGGER.info(
+        "AlterPipeProcedureV2: rollbackFromOperateOnDataNodes({})", alterPipeRequest.getPipeName());
+
+    // Push all pipe metas to datanode, may be time-consuming
+    String exceptionMessage =
+        parsePushPipeMetaExceptionForPipe(
+            alterPipeRequest.getPipeName(), pushPipeMetaToDataNodes(env));
+    if (!exceptionMessage.isEmpty()) {
+      LOGGER.warn(
+          "Failed to rollback alter pipe {}, details: {}, metadata will be synchronized later.",
+          alterPipeRequest.getPipeName(),
+          exceptionMessage);
+    }
+  }
+
+  @Override
+  public void serialize(DataOutputStream stream) throws IOException {
+    stream.writeShort(ProcedureType.ALTER_PIPE_PROCEDURE_V2.getTypeCode());
+    super.serialize(stream);
+    ReadWriteIOUtils.write(alterPipeRequest.getPipeName(), stream);
+    ReadWriteIOUtils.write(alterPipeRequest.getProcessorAttributesSize(), stream);
+    for (Map.Entry<String, String> entry : alterPipeRequest.getProcessorAttributes().entrySet()) {
+      ReadWriteIOUtils.write(entry.getKey(), stream);
+      ReadWriteIOUtils.write(entry.getValue(), stream);
+    }
+    ReadWriteIOUtils.write(alterPipeRequest.getConnectorAttributesSize(), stream);
+    for (Map.Entry<String, String> entry : alterPipeRequest.getConnectorAttributes().entrySet()) {
+      ReadWriteIOUtils.write(entry.getKey(), stream);
+      ReadWriteIOUtils.write(entry.getValue(), stream);
+    }
+    if (currentPipeStaticMeta != null) {
+      ReadWriteIOUtils.write(true, stream);
+      currentPipeStaticMeta.serialize(stream);
+    } else {
+      ReadWriteIOUtils.write(false, stream);
+    }
+    if (updatedPipeStaticMeta != null) {
+      ReadWriteIOUtils.write(true, stream);
+      updatedPipeStaticMeta.serialize(stream);
+    } else {
+      ReadWriteIOUtils.write(false, stream);
+    }
+    if (currentPipeRuntimeMeta != null) {
+      ReadWriteIOUtils.write(true, stream);
+      currentPipeRuntimeMeta.serialize(stream);
+    } else {
+      ReadWriteIOUtils.write(false, stream);
+    }
+    if (updatedPipeRuntimeMeta != null) {
+      ReadWriteIOUtils.write(true, stream);
+      updatedPipeRuntimeMeta.serialize(stream);
+    } else {
+      ReadWriteIOUtils.write(false, stream);
+    }
+  }
+
+  @Override
+  public void deserialize(ByteBuffer byteBuffer) {
```

Comment:
Consider adding a ser/de test as AlterPipeProcedureV2Test(can refer to CreatePipeProcedureV2Test)